server/src/modules/stock/stock.controller.ts

```diff
@@ -0,0 +1,102 @@
+import type { NextFunction, Request, Response } from "express"
+import { catchAsyncError } from "../../lib/catch-async-error"
+import { ErrorHandler } from "../../lib/error-handler"
+import {
+  createStockValidator,
+  deleteStockValidator,
+  updateStockValidator,
+} from "./stock.validator"
+import {
+  createStockService,
+  deleteStockService,
+  getStockService,
+  updateStockService,
+} from "./stock.service"
+
+export const createStockController = catchAsyncError(
+  async (req: Request, res: Response, next: NextFunction) => {
+    try {
+      const StockData = createStockValidator.parse(req.body)
+      try {
+        const Stock = await createStockService(StockData)
+        res.status(200).json({
+          success: true,
+          message: "Stock created successfully!",
+          data: Stock,
+        })
+      } catch (error) {
+        console.error(error)
+        next(new ErrorHandler("Stock coundn't created!", 500))
+      }
+    } catch (error) {
+      console.error(error)
+      next(new ErrorHandler("Invalid request body!", 400))
+    }
+  },
+)
+
+export const getStockController = catchAsyncError(
+  async (req: Request, res: Response, next: NextFunction) => {
+    try {
+      const StockData = await getStockService()
```

Comment:
We should update the variable name `StockData` to lowercase `stockData` here as well.

Reply:
done